### DISCUSSION TITLE
Custom Providers

### DISCUSSION THREAD

**Discussion by amcorch** [IEvangelist/azure-cosmos-dotnet-repository#272] [Ideas]
I would love to be able to create my own provider for some of the interfaces that are available. If the provider interfaces were public, I could create my own implementation for them, rather than using the default implementations.

My reason for doing this would be to support multiple cosmos db accounts (I would like to use this repository in a multi-tenant web api where some tenants would have their own accounts/databaes/containers etc).

Any help or suggestions would be appreciated.

**Comment by mumby0168** [comment]
Hi @amcorch we have taken the approach to keep everything internal as a first pass but we can certainly look at opening up the interfaces, which was did you want and what was the reason behind it?

We do have a discussion around multi tenancy/ multiple database support here #271
